### ISSUE TITLE
Social API must contain 443 port to work for modern team sites

### ISSUE THREAD

**Issue by MijoBarbaric** [pnp/pnpjs#593]
Reading out follow information for a modern team site via

```javascript
public async getIsSiteFollowed(siteUrl: string): Promise<boolean> {
    const siteActorInfo = {
        ActorType: SocialActorType.Site,
        ContentUri: siteUrl
    } as SocialActorInfo;

    let isFollowed = await sp.social.isFollowed(siteActorInfo);
    return isFollowed;
}
```
returns incorrect information. The site url (`webPartContext.pageContext.site.absoluteUrl`) can be something like `https://shspdevmijo.sharepoint.com/sites/MySiteName`. The social API works only with urls like `https://shspdevmijo.sharepoint.com:443/sites/MySiteName` (notice the https 443 port). This specific site url seems not to be available anywhere...

The same behaviour can be found when using `sp.social.follow()` and `sp.social.stopFollowing()`, respectively. Without 443-url they don't work as expected.

Currently we are doing a workaround by "correcting" the site url to contain this 443 port (hardcoded). Doesn't feel that future-proof, though. 😉 

### Category
- [ ] Enhancement
- [X] Bug
- [ ] Question
- [ ] Documentation gap/issue

pnp/sp version: 1.2.8
Environment: SharePoint Online
Modern team site

**Comment by patrick-rodgers** [comment]
This seems like a problem with the service? I am not sure we should apply a "fix" for this in the library as my tenant does not show this behavior and I am not sure others do.

**Comment by MijoBarbaric** [comment]
What kind of output do you get with this?

```javascript
let userFollowed = await sp.social.my.followed(
    SocialActorTypes.Site | SocialActorTypes.IncludeGroupsSites | SocialActorTypes.Document);

console.log("Following sites/documents:", userFollowed);
```
I've tested it on three different tenants and the biggest part of the resulting urls contain :443 port. I wonder when and why SharePoint adds this port to the url.

![image](https://user-images.githubusercontent.com/5558140/56132453-4c324800-5f8a-11e9-8e7f-a8ebdb321094.png)
 
I understand that this probably is no pnp library issue since you're doing nothing else than calling SharePoint APIs. But we don't receive any trustworthy site follow information because pnp compares non-443-urls with 443-urls and of course they don't match 1:1. 

So: How can we retrieve 443-url of the current site?

**Comment by patrick-rodgers** [comment]
Hi @MijoBarbaric - sorry for the delay in getting back to this one.

In my tenant I now see the urls come back with the 443 appended to the url, but I seem to be getting the right data. I didn't see that before so a change must have rolled out. Are you seeing different data depending on if the port is there or not?

Or is the issue that you then want to do stuff with those sites? I seem to get the same results, and I can then make calls to them with the port, that should work OK. When you say pnp compares them what do you mean?

In short, I see now this behavior in my tenant, but what problem do we need to solve? What is blocked or not working?

**Comment by patrick-rodgers** [comment]
Going to close this due to inactivity. If this is still an issue or there are blockers, please open a new issue and reference this one. Thanks!
